### PR TITLE
Azure combine move remove

### PR DESCRIPTION
**Pull request by wrp801**
Combined the following branches into one:
- sc-4044/azure-blob-storage-move-file-s
- sc-4045/azure-blob-storage-remove-file-s

### CODE REVIEW

File: azurestorage_blueprints/move_file.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+def find_matching_files(file_blobs, file_name_re):
+    """
+    Return a list of all file_names that matched the regular expression.
+    """
+    matching_file_names = []
+    for blob in file_blobs:
+        if re.search(file_name_re, blob.name):
+            matching_file_names.append(blob.name)
+
+    return matching_file_names
```
Comment (blakeburch) [inline azurestorage_blueprints/move_file.py]: Same comment as GCS. Use shipyard_utils files.find_all_file_matches() for consistency across Blueprints.

File: azurestorage_blueprints/move_file.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    copy_status_data = destination_blob.get_blob_properties().copy
+
+    if copy_status_data.status != "success":
+        # if copy failsm abort copy and return error message
+        destination_blob.abort_copy(copy_status_data.id)
+        print(f"Copy blob from {source_full_path} failed with status {copy_status_data.status}")
+        sys.exit(ec.EXIT_CODE_AZURE_MOVE_ERROR)
+    # successfully copied, so we delete the source path
+    source_blob.delete_blob()
+
```
Comment (blakeburch) [inline azurestorage_blueprints/move_file.py:108]: We need something in this function that prints a confirmation of the original path and the new path. Same thing as we had for S3/GCS

File: azurestorage_blueprints/delete_file.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+def azure_find_matching_files(file_blobs, file_name_re):
+    """
+    Return a list of all file_names that matched the regular expression.
+    """
+    matching_file_names = []
+    for blob in file_blobs:
+        if re.search(file_name_re, blob.name):
+            matching_file_names.append(blob.name)
+
+    return matching_file_names
```
Comment (blakeburch) [inline azurestorage_blueprints/delete_file.py:76]: Same comment as GCS. Use shipyard_utils files.find_all_file_matches() for consistency across Blueprints

**Review by blakeburch — CHANGES_REQUESTED** [review]
Minor tweaks needed.

Is there any reason we didn't build this the same as AWS/GCS where you could move the file between containers?

File: azurestorage_blueprints/delete_file.py (hunk 3 above)
Comment (wrp801) [reply azurestorage_blueprints/delete_file.py:76]: This is done now

File: azurestorage_blueprints/move_file.py (hunk 2 above)
Comment (wrp801) [reply azurestorage_blueprints/move_file.py:108]: Added in a message saying that the blob has been moved from `<original location>` to `<new location>`

File: azurestorage_blueprints/move_file.py (hunk 1 above)
Comment (wrp801) [reply azurestorage_blueprints/move_file.py]: This is done

**Review by blakeburch — APPROVED** [review]
Good to go.

**Pull request merged**